### PR TITLE
Added SDL_Mixer

### PR DESCRIPTION
**Pull request by LuukSantegoeds**
# Description
I implemented SDL_Mixer in the engine.

**MARK PLEASE READ THIS IN CASE YOU FORGOT**
The SoundManager is currently based on the principle of use just wanting to play some background music. Not 2 different tracks, not 3, just 1. Oh, and sound effects will need their own functions but we do not need those.

## Screenshot(s)
_I cannot visualize the sounds_

# How can this be tested?
Combine this PR with the PR from the [game](https://github.com/Brick-Studios/BeastArena/pull/38). The main menu and each level has its own music. Box.json proves that the music loops infinite.

- [x] Have ears
- [x] Play the game through all levels
- [x] Use the ears

**Please listen closely, the SoundManager inits SDL_Mixer with a frequency of 44100 but idk how that sounds your pc's**

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Included 1 or more screenshots
- [X] Code is const correct

### CODE REVIEW

File: include/brickengine/sound/sound_manager.hpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class SoundManager {
+public:
+    SoundManager();
+    ~SoundManager();
+
+    void playMusic(std::string path);
+    void pauseMusic(const bool pause) const;
+    void stopMusic();
+private:
+    Mix_Music *current_music = nullptr;
```
Comment (Bram-Boris) [inline include/brickengine/sound/sound_manager.hpp]: We do pointers like this `Mix_Music*`

File: lib/sound/sound_manager.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+void SoundManager::playMusic(std::string path) {
+    stopMusic();
+    
+    const std::string full_path = "assets/sound/" + path;
+    current_music = Mix_LoadMUS(full_path.c_str());
+    Mix_PlayMusic(current_music, -1); // The -1 is for infinite repeat
+}
+
+void SoundManager::pauseMusic(const bool pause) const {
```
Comment (Bram-Boris) [inline lib/sound/sound_manager.cpp]: This is more a toggle instead of pause

Comment (JanSchollaert) [reply lib/sound/sound_manager.cpp]: It should still be pause, the function to pause it is also called Mix_PauseMusic so imo it should stay like this,

Comment (Bram-Boris) [reply lib/sound/sound_manager.cpp]: Yes but that is only when the boolean is true/false. If you follow that logic it should be two seperate functions

**Pull request merged**